### DISCUSSION TITLE
`auth-helpers-nextjs` supabaseClient weird import behavior

### DISCUSSION THREAD

**Discussion by ryanccn** [supabase/supabase#8074] [Questions]
In one of my Next.js projects, I decided to use Supabase and `@supabase/auth-helpers-nextjs` for database and authentication. However, when fetching data from client side, I found that only importing `supabaseClient` from page components worked, and importing `supabaseClient` from a separate `lib/supabase.ts` resulted in the anon key being used instead of the user key.

For instance:

```typescript
// lib/supabase.ts

import { supabaseClient } from '@supabase/auth-helpers-nextjs';

const db = supabaseClient.from<definitions['mod_lists']>('mod_lists');

export const getUserLists = async () => {
  const ret = await db.select('*');

  if (ret.error) {
    console.error(ret.error);
    return [];
  }

  return ret.data;
};
```

```typescript
// pages/dashboard.tsx

const { username, isLoading, user } = useUserMore();
const [lists, setLists] = useState(null);

useEffect(() => {
  if (user) {
    getUserLists().then(setLists);
  }
}, [user]);
```

This does not work. While this *does*:

```diff
// lib/supabase.ts

- const db = supabaseClient.from<definitions['mod_lists']>('mod_lists');
+ const db = (client: SupabaseClient) =>
+  client.from<definitions['mod_lists']>('mod_lists');

- export const getUserLists = async () => {
-   const ret = await db.select('*');
+ export const getUserLists = async (
+   client: SupabaseClient
+ ): Promise<ModList[]> => {
+   const ret = await db(client).select('*');

// pages/dashboard.tsx

+ import { supabaseClient } from '@supabase/auth-helpers-nextjs';

- getUserLists().then(setLists);
+ getUserLists(supabaseClient).then(setLists);
```

Basically, by importing `supabaseClient` from the page component and passing it to the function, the authentication and RLS started working.

Is this intentional behavior? Because it is very painful to reimport in every page while I could have a shared library to use.

**Comment by ryanccn** [comment]
For reference, this is the commit:

https://github.com/ryanccn/moddermore/commit/ae5d2231d2c56a94080a52e9ec4850090c144bcb
